Snakefile

```diff
@@ -1,141 +1,163 @@
 import glob
 import os
 configfile: "./CONFIG.txt"
-chromosomes=config["chromosomes"]
-genome=config["genome_name"]
-
-array_positions = glob.glob(os.path.join("resources/snp_arrays", genome, "SNP_array_positions.txt"))[0]
 
-print(array_positions)
+chromosomes=config["chromosomes"]
+output_dir = config["output_dir"]
+temp_dir = config["temp_dir"] if config["temp_dir"] else output_dir + "/temp"
 
-cases=config["bams"].keys()
 
 rule all:
     input:
-        expand("output/{id}/cna.txt", id = cases)
-    
+        expand("{output_dir}/cna.txt", output_dir=output_dir)
+
+
+def devtools_install():
+    if config["ploidetect_local_clone"]:
+        devtools_cmd = "\"devtools::install_local('" + config["ploidetect_local_clone"] + "')\""
+    else:
+        devtools_cmd = "\"devtools::install_github('lculibrk/Ploidetect', "
+        devtools_cmd += "ref = '" + config["ploidetect_github_version"] + "')\""
+    return devtools_cmd
+
+rule install_ploidetect:
+    output:
+        "conda_configs/ploidetect_installed.txt"
+    conda:
+        "conda_configs/r.yaml"
+    params:
+        devtools_install()
+    shell:
+        "export LC_ALL=en_US.UTF-8; "
+        " Rscript -e {params} "
+        " && echo {params} > {output} && date >> {output}"
+
+
 rule germline_cov:
```

Comment:
besides trying to label 'temp()' on certain outputs, most of the changes formatting.